discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java

```diff
@@ -63,23 +64,30 @@
     private static final String EUREKA_CONFIG_SHOULD_REGISTER = "eureka.registration.enabled";
     private static final String EUREKA_CONFIG_METADATA_WEIGHT = "eureka.metadata.weight";
     private static final int EUREKA_REFRESH_INTERVAL = 5;
+    private static final int MAP_INITIAL_CAPACITY = 8;
     private static final String DEFAULT_WEIGHT = "1";
     private static final Configuration FILE_CONFIG = ConfigurationFactory.FILE_INSTANCE;
-    private static final ConcurrentMap<String, Set<InetSocketAddress>> CLUSTER_ADDRESS_MAP = new ConcurrentHashMap<>();
+    private static ConcurrentMap<String, Set<InetSocketAddress>> CLUSTER_ADDRESS_MAP;
 
     private static volatile boolean subscribeListener = false;
     private static volatile ApplicationInfoManager applicationInfoManager;
     private static volatile CustomEurekaInstanceConfig instanceConfig;
     private static volatile EurekaRegistryServiceImpl instance;
     private static volatile EurekaClient eurekaClient;
 
-    private EurekaRegistryServiceImpl() {}
+
+    private EurekaRegistryServiceImpl() {
+    }
 
     public static EurekaRegistryServiceImpl getInstance() {
         if (null == instance) {
             synchronized (EurekaRegistryServiceImpl.class) {
                 if (null == instance) {
                     instance = new EurekaRegistryServiceImpl();
+                    CLUSTER_ADDRESS_MAP = new ConcurrentHashMap<>(MAP_INITIAL_CAPACITY);
+                    instanceConfig = new CustomEurekaInstanceConfig();
+                    InstanceInfo instanceInfo = new EurekaConfigBasedInstanceInfoProvider(instanceConfig).get();
```

Comment:
The previous logic config is initialized, and then the instance is built. You are constructing an empty instance.

Reply:
my fault.i will fix it.

Reply:
done